tests/dataSources_test.go

```diff
@@ -36,6 +38,82 @@ var _ = Describe("DataSources", func() {
 	const cdiLabel = cdiLabelPrefix + "/dataImportCron"
 	const cdiCleanupLabel = cdiLabel + ".cleanup"
 
+	const registryURL = "docker://quay.io/kubevirt/cirros-container-disk-demo"
+
+	var commonAnnotations = map[string]string{
+		"cdi.kubevirt.io/storage.bind.immediate.requested": "true",
+	}
+
+	// Helper context that can be reused
+	var contextWithPvc = func(text string, args ...any) {
+		Context("with existing PVC", func() {
+			var (
+				dataVolume *cdiv1beta1.DataVolume
+			)
+
+			BeforeEach(func() {
+				dataVolume = &cdiv1beta1.DataVolume{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        dataSourceName,
+						Namespace:   internal.GoldenImagesNamespace,
+						Annotations: commonAnnotations,
+					},
+					Spec: cdiv1beta1.DataVolumeSpec{
+						Source: &cdiv1beta1.DataVolumeSource{
+							Registry: &cdiv1beta1.DataVolumeSourceRegistry{
+								URL:        ptr.To(registryURL),
+								PullMethod: ptr.To(cdiv1beta1.RegistryPullNode),
+							},
+						},
+						Storage: &cdiv1beta1.StorageSpec{
+							Resources: core.VolumeResourceRequirements{
+								Requests: core.ResourceList{
+									core.ResourceStorage: resource.MustParse("128Mi"),
+								},
+							},
+						},
+					},
+				}
+				Expect(apiClient.Create(ctx, dataVolume)).To(Succeed())
+
+				Eventually(func(g Gomega) {
+					foundDv := &cdiv1beta1.DataVolume{}
+					err := apiClient.Get(ctx, client.ObjectKeyFromObject(dataVolume), foundDv)
+
+					if errors.IsNotFound(err) {
+						g.Expect(apiClient.Get(ctx,
+							client.ObjectKeyFromObject(dataVolume),
+							&core.PersistentVolumeClaim{}),
+						).To(Succeed())
+						return
+					}
+
+					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(foundDv.Status.Phase).To(Equal(cdiv1beta1.Succeeded))
+				}, env.Timeout(), time.Second).Should(Succeed(), "DataVolume should successfully import.")
+			})
+
+			AfterEach(func() {
+				pvc := &core.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{Name: dataVolume.Name, Namespace: dataVolume.Namespace}}
+
+				errDv := apiClient.Delete(ctx, dataVolume)
+				errPvc := apiClient.Delete(ctx, pvc)
+
+				if !errors.IsNotFound(errDv) {
+					Expect(errDv).ToNot(HaveOccurred(), "Failed to delete data volume")
+				}
+				if !errors.IsNotFound(errPvc) {
+					Expect(errPvc).ToNot(HaveOccurred(), "Failed to delete persistent volume claim")
+				}
```

Comment:
Could also be an `Expect(err).To(Or(Not(HaveOccured(), MatchError(errors.IsNotFound, "errors.IsNotFound")))`

Reply:
Or perhaps
```
Expect(apiClient.Delete(ctx, dataVolume)).To(Or(Succeed(), MatchError(errors.IsNotFound, "errors.IsNotFound")))
Expect(apiClient.Delete(ctx, pvc)).To(Or(Succeed(), MatchError(errors.IsNotFound, "errors.IsNotFound")))
```